### PR TITLE
Migrated CollectionTask.Flag to Coroutines

### PR DESCRIPTION
**Pull request by divyansh-dxn**
* Post update flag behaviour has been taken up from FlagCardHandler which is just a subclass, without any additional implementations, of SuspendCardHandler.

* CollectionTask.Flag extends DismissNotes class which provides just a wrapper which allows Flag.actualTask to executeInTransaction. Same functionality has been combined and implemented in function updateSelectedCardsFlag.

IDK why the class is named DismissNotes, really not describes what it does.

Previous implementation of Flag was really messy and very hard to follow unless you've gone line by line code (took me enough time to understand). Currently it is single commit, and I couldn't figure out to break this commit into smaller ones. Will try though, if reviewing becomes harder.  I guess this migration makes it really easy to follow up what happens and where. 

## Fixes
Fixes a part of #7108

## How Has This Been Tested?
Unit tests and Pixel 4a API 32

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 1]
```diff
@@ -995,12 +995,41 @@ open class CardBrowser :
...
-    fun flagTask(flag: Int) {
-        TaskManager.launchCollectionTask(
-            CollectionTask.Flag(selectedCardIds, flag),
-            flagCardHandler()
-        )
+    suspend fun updateSelectedCardsFlag(flag: Int) {
+        val updatedCards = withProgress {
+            withCol {
+                val cards = selectedCardIds.map { getCard(it) }
+                col.db.database.beginTransaction()
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:1015]: The collection is already(from withCol) available so there's no need for `col.db....`.

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 2]
```diff
@@ -995,12 +995,41 @@ open class CardBrowser :
...
+                        c.load()
+                    }
+                    col.db.database.setTransactionSuccessful()
+                } finally {
+                    DB.safeEndInTransaction(col.db)
+                }
+                cards
+            }
+        }
+        updateCardsInList(updatedCards)
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:1028]: Add a todo above this method:

```
// try to offload the cards processing in updateCardsInList() on a background thread, 
// otherwise it could hang the main thread
```

**Review by lukstbit — CHANGES_REQUESTED** [review]
Looks ok with two minor issues.

**Review by lukstbit — APPROVED** [review]

**Review by ShridharGoel — APPROVED** [review]
